cmd/bucket-metadata-sys.go

```diff
@@ -55,13 +63,50 @@ func (sys *BucketMetadataSys) Remove(bucket string) {
 	sys.Unlock()
 }
 
+// DeleteBucketEtcdInformation deletes the bucket caching metainformation from etcd
+func (sys *BucketMetadataSys) DeleteBucketEtcdInformation(ctx context.Context, bucket string) {
+	if !globalIsGateway || globalGatewayName != NASXLBackendGateway || globalEtcdClient == nil {
+		return
+	}
```

Comment:
We decided to implement this feature (caching of bucket metadata) only for `nasxl` because we only saw a performance impact regarding bucket metadata in this mode. In `nasxl` mode the bucket metadata have to be read in order to check if versioning or locking is enabled. `nas` mode does not support versioning or locking. Thus reading bucket metadata for these cases is just a noop.

Reply:
Okay, understood. Can you add this to the commit message? You can just append your comment 1:1, it's a good explanation.